### ISSUE TITLE
gmsv_workshop_linux.dll - attempt to index global 'steamworks' (a nil value)

### ISSUE THREAD

**Issue by MrCorn812** [WilliamVenner/gmsv_workshop#8]
I am running a x32 server and am attempting to use steamworks but am having difficulties with it. I have added it to the correct path and I also went through the troubleshooting steps but have had no luck as of yet. 

I was wondering if I was able to get some help with this. I have also sent you an email.

[The error.](https://i.gyazo.com/c35f74a20b5612fd0668e3639d3aa904.png)

![image](https://github.com/WilliamVenner/gmsv_workshop/assets/150591628/35755a12-73d8-48b6-87d5-6c7c9d2a0125)

![image](https://github.com/WilliamVenner/gmsv_workshop/assets/150591628/ef7a0fd7-bb52-4463-a471-a6904572fa6a)

**Comment by WilliamVenner** [comment]
1. Did you put `require("workshop")` in a serverside file?
2. Is it being called BEFORE line 144 in sv_workshop.lua?
3. Did you install the DLL to garrysmod/lua/bin/gmsv_workshop_linux.dll ?
4. Are there any errors when the DLL is `require`d?

**Comment by MrCorn812** [comment]
1. Yes
2. Yes
3. Yesssir
4. No errors appear when workshop is being required

**Comment by MrCorn812** [comment]
Any update regarding this?

**Comment by FlorianLeChat** [comment]
I've just tested this on my side just in case function signatures or anything like that have changed with the latest Garry's Mod updates, but everything works fine.
- Are you using a provider for your server? (If yes, which one? If not, what's your operating system?)
- Could you show the content of your file to check how the module is loaded?
- Could you write `status` in the server console and show what it contains?

**Comment by MrCorn812** [comment]
Yeah its weird how its not working for me not sure why, it worked on another dedi we had.

We are  just running the server through pterodactyl (Ubuntu 22.04).
[.dll location](https://i.gyazo.com/a057bd1c948de1460be4d7dbf170a6a4.png)
Our [server ](https://i.gyazo.com/48daac935cc703fc0eead644bea1a9ab.png)is on **version : 2023.06.28/24 9040 secure** (I have tried using older versions of the module)

**Comment by FlorianLeChat** [comment]
Weird, there's nothing special. My dedicated server is running under Docker with Debian 12 with the same version as yours. There's either an issue with this Lua file (unless an error also occurs if you run the function with `lua_run` in the server console, `lua_run require("workshop") steamworks.<function>()...`) or with the module itself, but only Billy will be able to help you further unfortunately.

**Comment by wubman22** [comment]
yer mate

**Comment by MrCorn812** [comment]
Yeah, I still get an error when I use lua_run, says the same thing. Hopefully we are able to get it working.
